func-complete-check/check-marker/check-marker.v1.ts

```diff
@@ -0,0 +1,176 @@
+import * as RA from 'ramda-adjunct'
+import * as R from 'ramda'
+import { IAsyncTableService, AsyncTableService } from '../lib/storage-helper'
+import { ValidatedCheck } from '../typings/message-schemas'
+import moment from 'moment'
+import { ICheckFormService, CheckFormService } from '../lib/check-form.service'
+import { ILogger } from '../lib/ILogger'
+
+export interface ICheckMarkerFunctionBindings {
+  receivedCheckTable: Array<any>
+  checkNotificationQueue: Array<any>
+}
+
+declare interface Answer {
+  factor1: number
+  factor2: number
+  answer: string
+  sequenceNumber: number
+  question: string
+  clientTimestamp: string
+}
+
+declare interface CheckFormQuestion {
+  f1: number
+  f2: number
```

Comment:
A question has only factor1 and factor2, but a CheckFormQuestion should also have a sequence - e.g. a CheckForm is a series of questions asked in a particular order.

Reply:
raw JSON from checkForm.formData...

```
[ 
        {
            "f1" : 2,
            "f2" : 5
        }, 
        {
            "f1" : 11,
            "f2" : 2
        }, 
        {
            "f1" : 5,
            "f2" : 10
        }, 
        {
            "f1" : 4,
            "f2" : 4
        }, 
        {
            "f1" : 3,
            "f2" : 9
        }, 
        {
            "f1" : 2,
            "f2" : 4
        }, 
        {
            "f1" : 3,
            "f2" : 3
        }, 
        {
            "f1" : 4,
            "f2" : 9
        }, 
        {
            "f1" : 6,
            "f2" : 5
        }, 
        {
            "f1" : 12,
            "f2" : 12
        }
    ]
```

Reply:
any enhancements to this can be formulated into the ticket i created and shared with you earlier.